server/src/main/java/io/crate/types/ResultSetParser.java

```diff
@@ -102,7 +104,9 @@ public static Object getObject(ResultSet resultSet, int i, String typeName) thro
             }
             case "json", "jsonb":
                 String json = resultSet.getString(columnIndex);
-                value = jsonToObject(json);
+                Map<String, Object> o = (Map<String, Object>) jsonToObject(json);
+                // This is hacky just to test the one level subscript
+                value = o.get(ref.column().path().getFirst());
```

Comment:
I know it's a temporal thing. Question rather about future - can we reuse some bits of the `SourceLookup` here? Similar to what we do in `DocRefResolver`.

Reply:
It would make sense yes. Ideally we should use the parsing once for all columns in the select list, and then just choose the sub-object we want, so if we have for example:
```
select o['x'], o['y'], o['x']['xi'] from foreign_table ...
```
we don't get to parse 3 times the object, but this can be optimized as a followup, imho.

Reply:
I think we have smth like that (`registerRef`) and can maybe also re-use. 
But yeah, that can be done in a separate PR